File: js/src/maplestory/api/common/dto/character/characterSkill.ts
export abstract class CharacterSkillDto {
  public abstract date: Date | null;
  public abstract characterClass: string | null;
  public abstract characterSkillGrade: string | null;
  public abstract characterSkill: CharacterSkillInfoDto[];
}

export abstract class CharacterSkillInfoDto {
  public abstract skillName: string;
  public abstract skillDescription: string;
  public abstract skillLevel: number;
  public abstract skillEffect: string;
  public abstract skillEffectNext: string | null;
  public abstract skillIcon: string;
}

File: csharp/MapleStory.OpenAPI/Src/Common/DTO/Character/CharacterSkill.cs
namespace MapleStory.OpenAPI.Common.DTO
{
    public abstract class CharacterSkillDTO<TCharacterSkillInfo>
    {
        public abstract DateTimeOffset? Date { get; set; }
        public abstract string CharacterClass { get; set; }
        public abstract string? CharacterSkillGrade { get; set; }
        public abstract List<TCharacterSkillInfo> CharacterSkill { get; set; }
    }

    public abstract class CharacterSkillDTO : CharacterSkillDTO<CharacterSkillInfoDTO>
    {

    }

    public abstract class CharacterSkillInfoDTO
    {
        public abstract string SkillName { get; set; }
        public abstract string SkillDescription { get; set; }
        public abstract long? SkillLevel { get; set; }
        public abstract string? SkillEffect { get; set; }
        public abstract string? SkillEffectNext { get; set; }
        public abstract string SkillIcon { get; set; }
    }
}
